consumer.go

```diff
@@ -241,20 +241,36 @@ func (c *Consumer) handleRecord(shardID string, r *Record, fn func(*Record) Scan
 }
 
 func (c *Consumer) getShardIDs(streamName string) ([]string, error) {
-	resp, err := c.client.DescribeStream(
-		&kinesis.DescribeStreamInput{
-			StreamName: aws.String(streamName),
-		},
-	)
-	if err != nil {
-		return nil, fmt.Errorf("describe stream error: %v", err)
-	}
-
 	var ss []string
-	for _, shard := range resp.StreamDescription.Shards {
-		ss = append(ss, *shard.ShardId)
+	var exclusiveStartShardId *string
+	for {
+		resp, err := c.client.DescribeStream(
+			&kinesis.DescribeStreamInput{
+				StreamName:            aws.String(streamName),
+				ExclusiveStartShardId: exclusiveStartShardId,
+			},
+		)
+		if err != nil {
+			return nil, fmt.Errorf("describe stream error: %v", err)
+		}
+
+		streamDescription := resp.StreamDescription
+		shards := streamDescription.Shards
+
+		if len(shards) == 0 {
+			return ss, nil
+		}
+
+		for _, shard := range shards {
+			ss = append(ss, *shard.ShardId)
+		}
+
+		exclusiveStartShardId = shards[len(shards)-1].ShardId
```

Comment:
wondering if `NextToken` might be better for itterator?

> // When the number of shards in the data stream is greater than the default
    // value for the MaxResults parameter, or if you explicitly specify a value
    // for MaxResults that is less than the number of shards in the data stream,
    // the response includes a pagination token named NextToken. You can specify
    // this NextToken value in a subsequent call to ListShards to list the next
    // set of shards.

Reply:
or maybe: 
https://godoc.org/github.com/aws/aws-sdk-go/service/kinesis#Kinesis.DescribeStreamPages

Reply:
something like this:

```go
func (c *Consumer) getShardIDs(streamName string) ([]string, error) {
	var ss []string

	err := c.client.DescribeStreamPages(
		&kinesis.DescribeStreamInput{
			StreamName: aws.String(streamName),
		},
		func(page *kinesis.DescribeStreamOutput, lastPage bool) bool {
			for _, shard := range page.StreamDescription.Shards {
				ss = append(ss, *shard.ShardId)
			}
			return true
		})

	return ss, err
}
```